### DISCUSSION TITLE
How can I have different levels for the different modules?

### DISCUSSION THREAD

**Discussion by ybbh** [tokio-rs/tracing#2412] [Q&A] [answered]
How can I have different levels for the different modules?
Suppose I have mod M and N, mod M imports codes from mod N.
I want to set M with a TRACE level and N with an INFO level.
Is there a simple approach to achieving this?

Thanks

**Comment by hawkw** [comment, marked as answer]
You can do this using a filter that enables different levels for each module you want to see logs from. See https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html for documentation on how to configure this kind of filtering.

**Reply by ybbh** [reply]
Thank you for your answer.
